### PR TITLE
Cleanup storage related deprecated flags and default feature gate that marked as remove in 1.23

### PR DESCRIPTION
**Pull request by niulechuan**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
In 1.23, some storage related default feature gate and flags should be remove, so we 
- Remove `CSIServiceAccountToken` from default feature gate map that should remove in 1.23
- Remove `experimental-mounter-path` and `experimental-check-node-capabilities-before-mount` from kubelet flags that will  deprecated in 1.23

#### Does this PR introduce a user-facing change?

```release-note
1. `CSIServiceAccountToken` feature gate will not enable by default
2. `experimental-mounter-path` flag will be removed
3. `experimental-check-node-capabilities-before-mount` flag will be removed
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -822,7 +822,6 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	ConfigurableFSGroupPolicy:                      {Default: true, PreRelease: featuregate.Beta},
 	CSIInlineVolume:                                {Default: true, PreRelease: featuregate.Beta},
 	CSIStorageCapacity:                             {Default: true, PreRelease: featuregate.Beta},
-	CSIServiceAccountToken:                         {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.23
```
Comment (pacoxu) [inline pkg/features/kube_features.go:825]: /cc @zshihang 
as the owner of CSIServiceAccountToken

Comment (zshihang) [reply pkg/features/kube_features.go:825]: lgtm

**Comment by pacoxu** [comment]
Fix the CI failure of pull-kubernetes-e2e-gce-alpha-features in https://github.com/kubernetes/test-infra/pull/23324

**Comment by pacoxu** [comment]
/lgtm cancel

cluster/gce/util.sh is using the experimental flags. The file below need to be updated for the e2e test failures.

https://github.com/kubernetes/kubernetes/blob/818ed1afff306f8b310c56083e0c6be4fa93ce8a/cluster/gce/util.sh#L760-L762

**Comment by dims** [comment]
Is this PR still needed, please rebase if so (or we can close it?)

**Comment by dims** [comment]
This PR is older than 4 weeks and needs as rebase.

Please rebase the PR against latest master and reopen if still needed.

/close

**Pull request closed without merging**